examples/test_api.c

```diff
@@ -277,42 +289,87 @@ double getbw(char* name, char* buf, size_t size, int times)
         strcpy(file, newname);
       }
 
-      /* open the file and write the checkpoint */
-      int fd_me = open(file, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
+      int fd_me;
+
+      if (use_shared_file) {
+        /* open the file and write the checkpoint */
+        if (rank == 0) {
+          fd_me = open(file, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
+          if (fd_me > 0) {
+            if (ftruncate(fd_me, total_filesize) < 0) {
+              printf("%d: Could not truncate file %s\n", rank, file);
+              close(fd_me);
+              fd_me = -1;
+            }
+            else {
+              if (lseek(fd_me, total_filesize-1, SEEK_SET) >= 0) {
+                char c = 'X';
+                if (write(fd_me, &c, 1) < 0) {
+                  printf("%d: Failed to resize file %s\n", rank, file);
+                  close(fd_me);
+                  fd_me = -1;
+                }
+              }
+              else {
+                  printf("%d: Failed to seek to 0x%08lx in file %s for resize\n", rank, total_filesize-1, file);
+                  close(fd_me);
+                  fd_me = -1;
+              }
+            }
```

Comment:
Just to simplify the code, do you mind if we drop this?  I think it should still work without it.